tests/globus_test.py

```diff
@@ -0,0 +1,394 @@
+
+                for checksum_type, value in checksum_map_filename1.items():
+                    # send checksum request via telnet, read response and validate checksums
+                    session.write(get_telnet_checksum_command(checksum_type, filename1))
+                    checksum = parse_telnet_checksum_response_as_string(session.read_until(b'\r\n'))
+                    self.assertEqual(checksum, value, f'{checksum_type} checksums are not equal')
+
+                    # request checksum again and make sure it returns quickly since it is cached in metadata
+                    time1 = time.time()
+                    session.write(get_telnet_checksum_command(checksum_type, filename1))
+                    checksum = parse_telnet_checksum_response_as_string(session.read_until(b'\r\n'))
+                    time2 = time.time()
+                    self.assertEqual(checksum, value, f'{checksum_type} checksums are not equal')
+                    self.assertLess(time2 - time1, 1, f'{checksum_type} checksums took too long to retrieve')
+
+            # overwrite filename1 with filename2
+            with FTP() as ftp:
+                ftp.connect(host='localhost', port=2811)
+                ftp.login(user='user1', passwd='pass')
+                with open(filename2,'rb') as f:
+                    ftp.storbinary(f'STOR /tempZone/home/user1/{filename1}', f)
+
+            # get and verify checksums.  the cached value in metadata should be discarded and checksums recalculated
+            with telnetlib.Telnet('localhost', 2811, 100) as session:
+                session.read_until(b'ready.\r\n')
+                session.write(b'user user1\r\n')
+                session.read_until(b'Password required for user1.\r\n')
+                session.write(b'pass\r\n')
+                session.read_until(b'User user1 logged in.\r\n')
+
+                for checksum_type, value in checksum_map_filename2.items():
+                    # send checksum request via telnet, read response and validate checksums
+                    session.write(get_telnet_checksum_command(checksum_type, filename1))
+                    checksum = parse_telnet_checksum_response_as_string(session.read_until(b'\r\n'))
+                    self.assertEqual(checksum, value, f'{checksum_type} checksums are not equal')
+
+        finally:
+            os.remove(filename1)
+            os.remove(filename2)
+
+            # remove file via ftp 
+            with FTP() as ftp:
+                ftp.connect(host='localhost', port=2811)
+                ftp.login(user='user1', passwd='pass')
+                ftp.delete(f'/tempZone/home/user1/{filename1}')
+
+    def test_checksum_with_progress_markers(self):
+        filename = f'{inspect.currentframe().f_code.co_name}1'
+
+        try:
+            # file must be big enough so that checksum takes at least one second
+            # otherwise this test will hang waiting for checksum marker
```

Comment:
Would this hang be referring to the `while` loop below? Perhaps we could build in a timeout and a delay between checks, a la `delayAssert`? I suppose all that would do is introduce an intermittent failure rather than an intermittent hang... Is there anything we can do here to avoid the timing in the test, or is that just how it has to be (for now)?

Reply:
With the latest change it will not hang.  I will remove that comment.

Reply:
Updated the comment to state that the test will simply fail if we don't receive a marker.  I did verify this by using a small file.